### PR TITLE
Update the css-conditional IDL file.

### PR DESCRIPTION
**Pull request by lukebjerring**
Hello reviewer(s),

This PR is intended to consolidate the spec’s IDL definition with the WPT test suite’s copy, and any idlharness tests.

The up-to-date copy of the IDL file was automatically extracted from the reffy-reports repo (https://github.com/tidoust/reffy-reports/tree/master/whatwg/idl) which scrapes known specs automatically + regulary.

This PR is part of a migration project which will eventually be automatically updating and creating PRs for changes in spec IDL.

Please check that:
The spec (and its source) is correct and up-to-date
All tests which cover the IDL in the spec have been migrated to fetch + use the idl in the `interfaces/` directory (instead of inline copies in the test files).

### CODE REVIEW

**Comment by gsnedders** [comment]
@lukebjerring why is this using the levelled spec?

**Comment by lukebjerring** [comment]
@gsnedders could you clarify what you mean by the levelled spec, and what you would expect to be used instead?

**Comment by gsnedders** [comment]
@lukebjerring we've gone from using https://drafts.csswg.org/css-conditional/ (without level in the URL) to https://drafts.csswg.org/css-conditional-3/ (with level 3 in the URL), and renamed the file per the old levelled format (to css3-conditional).

**Comment by lukebjerring** [comment]
Filed https://github.com/tidoust/reffy-reports/issues/11

**Comment by lukebjerring** [comment]
Renamed back to css-conditional. PTAL.

File: css/css-conditional/idlharness.html [hunk 1]
```diff
@@ -14,18 +14,31 @@
...
+        idl_array.add_objects({
+          CSSRule: ['rule'],
+          CSSMediaRule: ['media'],
+          CSSSupportsRule: ['supports'],
+        });
+        idl_array.prevent_multiple_testing('CSSRule');
+
+        self.media = document.styleSheets[0].cssRules[0];
+        self.supports = document.styleSheets[0].cssRules[1];
+        self.rule = media.cssRules[0];
```
Comment (foolip) [inline css/css-conditional/idlharness.html]: Does this actually work, is this object a CSSRule and not some more specific type?

File: interfaces/css-conditional.idl [hunk 2]
```diff
@@ -27,7 +28,7 @@ interface CSSMediaRule : CSSConditionRule {
 interface CSSSupportsRule : CSSConditionRule {
 };
 
-partial interface CSS {
-  static boolean supports(CSSOMString property, CSSOMString value);
-  static boolean supports(CSSOMString conditionText);
+partial namespace CSS {
```
Comment (foolip) [inline interfaces/css-conditional.idl]: cssom.idl still has interface CSS, so this will break, right?

**Review by foolip — COMMENTED** [review]
Fixed two little things and wonder about more, can you run and verify this still works?

File: interfaces/css-conditional.idl (hunk 2 above)
Comment (lukebjerring) [reply interfaces/css-conditional.idl]: Not any more :)

**Comment by lukebjerring** [comment]
Note that this has a single failing test because the IDL in WPT says that `CSSConditionRule.conditionText` is not readonly, but Chrome's currently IDL says it is readonly.

https://cs.chromium.org/chromium/src/third_party/blink/renderer/core/css/css_condition_rule.idl?l=8&rcl=0f0fa2e588f06f76c425651f608fe0e50c80e3ef

**Review by foolip — APPROVED** [review]
Tested in Chrome and Firefox. A different test fails in Firefox, looks like `insertRule` has a 2nd optional argument which shouldn't be optional.

**Pull request merged**